src/coreComponents/physicsSolvers/multiphysics/PoromechanicsKernelBase.hpp

```diff
@@ -0,0 +1,266 @@
+    /// Derivative of total stress wrt pressure
+    real64 dTotalStress_dPressure[6]{};
+    /// Derivative of total stress wrt temperature
+    real64 dTotalStress_dTemperature[6]{};
+
+    /// Body force
+    real64 bodyForce[3]{};
+    /// Derivative of body force wrt volumetric strain increment
+    real64 dBodyForce_dVolStrainIncrement[3]{};
+    /// Derivative of body force wrt pressure
+    real64 dBodyForce_dPressure[3]{};
+
+    /// Delta temperature since the beginning of the simulation
+    real64 deltaTemperatureFromInit{}; // for stress computation
+    /// Delta temperature since last time step
+    real64 deltaTemperatureFromLastStep{}; // for porosity update
+
+    // Storage for residual and degrees of freedom
+
+    /// Linear momentum balance residual
+    real64 ( &localResidualMomentum )[numDispDofPerElem];
+    /// Derivative of linear momentum balance residual wrt displacement
+    real64 ( &dLocalResidualMomentum_dDisplacement )[numDispDofPerElem][numDispDofPerElem];
+    /// Derivative of linear momentum balance residual wrt pressure
+    real64 dLocalResidualMomentum_dPressure[numDispDofPerElem][1]{};
+
+    /// C-array storage for the element local row degrees of freedom.
+    globalIndex localPressureDofIndex[1]{};
+
+  };
+  //*****************************************************************************
+
+  /**
+   * @brief Copy global values from primary field to a local stack array.
+   * @copydoc ::geosx::finiteElement::ImplicitKernelBase::setup
+   *
+   * For the PoromechanicsKernelBase implementation, global values from the displacement,
+   * incremental displacement, and degree of freedom numbers are placed into
+   * element local stack storage.
+   */
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void setup( localIndex const k,
+              StackVariables & stack ) const
+  {
+    for( localIndex a = 0; a < numNodesPerElem; ++a )
+    {
+      localIndex const localNodeIndex = m_elemsToNodes( k, a );
+
+      for( integer i = 0; i < 3; ++i )
+      {
+#if defined(CALC_FEM_SHAPE_IN_KERNEL)
+        stack.xLocal[a][i] = m_X[localNodeIndex][i];
+#endif
+        stack.u_local[a][i] = m_disp[localNodeIndex][i];
+        stack.uhat_local[a][i] = m_uhat[localNodeIndex][i];
+        stack.localRowDofIndex[a*3+i] = m_dofNumber[localNodeIndex]+i;
+        stack.localColDofIndex[a*3+i] = m_dofNumber[localNodeIndex]+i;
+      }
+    }
+
+    stack.localPressureDofIndex[0] = m_flowDofNumber[k];
+  }
+
+protected:
+
+  /// The array containing the nodal position array.
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const m_X;
+
+  /// The rank-global displacement array.
+  arrayView2d< real64 const, nodes::TOTAL_DISPLACEMENT_USD > const m_disp;
+
+  /// The rank-global incremental displacement array.
+  arrayView2d< real64 const, nodes::INCR_DISPLACEMENT_USD > const m_uhat;
+
+  /// The gravity vector.
+  real64 const m_gravityVector[3]{};
+  real64 const m_gravityAcceleration;
```

Comment:
Can you explain the difference those two vectors?

Reply:
Done, I added a comment to explain the difference between these two member variables.